roles/bmc_fw_update/tasks/main.yml

```diff
@@ -1,2 +1,85 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022-2024 Dell Inc, or its subsidiaries.
+
 ---
 # tasks file for bmc_fw_update
+
+- name: Get Firmware Inventory
+  community.general.redfish_info:
+    category: Update
+    command: GetFirmwareInventory
+    baseuri: "{{ inventory_hostname }}"
+    username: "{{ dpu_bmc_username }}"
+    password: "{{ dpu_bmc_password }}"
+  register: fw_inventory_before
+
+- name: Print BMC Version
+  ansible.builtin.debug:
+    msg: "{{ fw_inventory_before.redfish_facts.firmware.entries[0].Version }}"
+
+- name: Download firmware {{ firmware.bmc }} image
+  ansible.builtin.get_url:
+    url: "{{ firmware.url }}/{{ firmware.bmc }}"
+    dest: /tmp/{{ firmware.bmc }}
+    mode: '0440'
+
+- name: Update BMC firmware of DPU
+  community.general.redfish_command:
+    category: Update
+    command: MultipartHTTPPushUpdate
+    baseuri: "{{ inventory_hostname }}"
+    username: "{{ dpu_bmc_username }}"
+    password: "{{ dpu_bmc_password }}"
+    timeout: 600
+    update_image_file: "/tmp/{{ firmware.bmc }}"
+  register: result_update_task
+
+- name: Print TASK id for tracking
```

Comment:
Printing the values will not help the user to get this at later stage, good to add  under the facts ex: role_name_facts.Taskid , so that in case the job is not finished even after the retries/delay. user has an hold of the task id to take up next action.